### DISCUSSION TITLE
2021 Feature Requests

### DISCUSSION THREAD

**Discussion by lukebp** [decred/politeiagui#2578] [Ideas]
This discussion will server as a general purpose feature request thread.

**Comment by lukebp** [comment]
**Feature request** 
Show the version of the record a comment was made on.
This was originally discussed in #1370.

**Comment by lukebp** [comment]
**Feature request**
Filtering by proposal domain.
This was originally discussed in #2069.

**Comment by lukebp** [comment]
**Feature request**
Make the diff viewer generic.
Originally discussed in #2537.

**Reply by lukebp** [reply]
Another issue with the current diff viewer is that inline images are not displayed in the diff.  Changes to the inline image filename is shown, but not the image itself.  We need to think about the best way to show these inline image diffs.

**Comment by victorgcramos** [comment]
one cool feature that we could implement with the new billing status changes is the proposal timeline

**Reply by lukebp** [reply]
What kind of proposal timeline?

**Reply by victorgcramos** [reply]
hey  Luke, sorry for answering 14 days later lol. Maybe show the date when some proposal has been: 1. published; 2. authorized; 3. started voting; 4. approved; 5. completed... I was brainstorming something like this:

 ![Proskynete/vertical-timeline-component-react](https://camo.githubusercontent.com/cf2e7b0ab760940d745dc552ed988e86422edbae0e92fda4f21d943c7499d2ba/68747470733a2f2f692e696d6775722e636f6d2f4b42714c496c4b2e706e67)

**Reply by lukebp** [reply]
Nice.  It would be helpful to include all major proposal state changes so you can see exactly how the proposal process unfolded.

**Reply by xaur** [reply]
I described it back in 2018: https://github.com/decred/politeiagui/issues/820 . The issue also links design mockups by Maria that may still be useful. Note that showing state changes that happened _before_ published is useful to make admins more accountable. I'm not sure it was on the mockups, but if it weren't, I would also show what user (including admins or Pi itself) triggered each change.

On a side note, [closing](https://github.com/decred/politeiagui/issues/820#issuecomment-485532217) with a redirection to dig through a huge design meta issue was not fun and broke a handy subscription to a specific feature for all participants.

Glad it's back on the radar.

**Comment by lukebp** [comment]
politeia comment tipping via lightning could help solve the issue of low quality comments creating too much noise.  Sorting by upvotes/downvotes works pretty well, but sorting by the amount of tip received is even costlier and would provide an even higher signal.

**Reply by xaur** [reply]
Previous 15-comment discussion here: https://github.com/decred/politeia/issues/899

The most fun insight I would copy from there is that tips could stimulate quality comments to appear sooner. For example, a user commenting with a thorough research or pros/cons analysis would help voters make a better decision, faster. It would also help authors to improve their proposals. A win for everyone.

Two issues that come to mind though are implementation complexity and privacy considerations (e.g. I think it should be hard to trace who tipped what).

**Comment by xaur** [comment]
**Global Activity Feed**

I start my Reddit day with discovering new activity using the subreddit's comments feed like [this](https://www.reddit.com/r/decred/comments/). It allows to never miss a comment and quickly see unprocessed things by the lack of upvote or downvote reactions.

A related feature is personalized activity feeds (aka subscriptions - inbox on Reddit and the bell icon on GitHub) but I think the global feed is easier to implement as a first step. I drafted some abstractions that might inspire an implementation in https://github.com/decred/politeia/issues/736 .

Originally discussed in https://github.com/decred/politeiagui/issues/783 .

**Reply by xaur** [reply]
I forgot to add that the feed of all comments also helps with moderation, I can filter out bad comments even when many threads are discussed in parallel.
